lib/g5kchecks/ohai/gpu.rb

```diff
@@ -4,39 +4,38 @@
 #gpu_output = "05:00.0 3D controller: NVIDIA Corporation GK210GL [Tesla K80] (rev a1)\n06:00.0 3D controller: NVIDIA Corporation GK210GL [Tesla K80] (rev a1)\n"
 #gpu_output = "03:00.0 3D controller: NVIDIA Corporation Device 17fd (rev a1)"
 
-if gpu_output != ''
-  gpu_lines = gpu_output.split("\n")
-  gpu_count = 0
-  gpu_vendor = 'Unknown'
-  gpu_model = 'Unknown'
+if gpu_output['NVIDIA']
+  nvidia_output = `nvidia-smi -L`
+  gpu_info_str = nvidia_output.split("\n")[-1].split()
+  gpu_model = gpu_info_str[3]
+  gpu_count = gpu_info_str[1].sub ':', ''
```

Comment:
`sub` replaces only the first occurrence; `tr` (as you used lower in the file) will replace all. Just checking in case it's not what you want.

Reply:
This is just to parse the GPU number from the output string. There's only one colon--'GPU X:'--that isn't handled by the initial split.